src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,148 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './TodoList.scss';
+import classNames from 'classnames';
+import { Todo } from '../../types/TodoType';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
+type Props = {
+  todos: Todo[],
+  chooseUserId: (userId: number) => void,
+  selectedUserId: number,
+};
+
+export const TodoList: React.FC<Props> = ({
+  todos,
+  chooseUserId,
+  selectedUserId,
+}) => {
+  const [query, setQuery] = useState('');
+  const [status, setStatus] = useState('all');
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+
+  useEffect(() => {
+    setVisibleTodos(todos);
+  }, [todos]);
+
+  const handlerSelect = (cell: string) => {
+    setStatus(cell);
+
+    let selectedTodos = todos;
+
+    if (cell === 'completed') {
+      selectedTodos = selectedTodos
+        .filter((selectedTodo) => selectedTodo.completed === true);
+    }
+
+    if (cell === 'active') {
+      selectedTodos = selectedTodos
+        .filter((selectedTodo) => selectedTodo.completed === false);
+    }
+
+    if (cell === 'all') {
+      selectedTodos = todos;
+    }
+
+    if (query.trim() !== '') {
+      selectedTodos = selectedTodos.filter((selectedTodo) => (
+        selectedTodo.title.includes(query.trim())
+      ));
```

Comment:
it would be good to combine checks by status and query so you don't need to filter twice